gae-interop-testing/gae-jdk7/src/main/java/io/grpc/testing/integration/OkHttpClientInteropServlet.java

```diff
@@ -49,6 +53,42 @@
 public final class OkHttpClientInteropServlet extends HttpServlet {
   private static final String INTEROP_TEST_ADDRESS = "grpc-test.sandbox.googleapis.com:443";
 
+  private static final class LogEntryRecorder extends Handler {
+    private List<LogRecord> loggedMessages = new ArrayList<LogRecord>();
+
+    @Override
+    public void publish(LogRecord logRecord) {
+      loggedMessages.add(logRecord);
+    }
+
+    @Override
+    public void flush() {}
+
+    @Override
+    public void close() {}
+
+    public String getLogOutput() {
+      SimpleFormatter formatter = new SimpleFormatter();
+      StringBuilder sb = new StringBuilder();
+      for (LogRecord loggedMessage : loggedMessages) {
+        sb.append(formatter.format(loggedMessage));
+      }
+      return sb.toString();
+    }
+  }
+
+  private final LogEntryRecorder handler = new LogEntryRecorder();
+
+  @Override
+  public void init() {
+    Logger.getLogger("").addHandler(handler);
```

Comment:
This is init for the servlet. Surely a new servlet isn't created for each request, right? Keeping all logs all the time is worrisome. Maybe just do this within the request?

Reply:
I was thinking this is fine for the narrow use case of this servlet, but now I realize the HTTP request may be retried and we don't care about previous entries. I'll do it naively here instead of using the fancier servlet APIs.